src/Powercord/plugins/pc-reactdevtools/index.js

```diff
@@ -1,60 +1,76 @@
-const { resolve, join } = require('path');
-const { remote } = require('electron');
-const { readdir } = require('fs').promises;
-const { Plugin } = require('powercord/entities');
+const { join } = require("path");
+const { remote } = require("electron");
+const { createWriteStream, existsSync } = require("fs");
+const { Plugin } = require("powercord/entities");
+const request = require("request");
+const unzip = require("unzip-crx");
 
 // stolen from https://github.com/Inve1951/BetterDiscordStuff/blob/master/plugins/enableReactDevtools.plugin.js
 module.exports = class ReactDevtools extends Plugin {
-  get path () {
-    const path = 'Default/Extensions/fmkadmapgofadopljbjfkapdkoienihi';
-    switch (process.platform) {
-      case 'win32':
-        return resolve(process.env.LOCALAPPDATA, 'Google/Chrome/User Data', path);
-      case 'linux':
-        return resolve(process.env.HOME, '.config/google-chrome', path);
-      case 'darwin':
-        return resolve(process.env.HOME, 'Library/Application Support/Google/Chrome', path);
-    }
+  get path() {
+    return join(__dirname, "rdt.crx");
+  }
 
-    return null;
+  get folderPath() {
+    return join(__dirname, "react-dev-tools");
   }
 
-  startPlugin () {
-    /*
-     * We should download our own version of react devtools and use it instead of pulling it from Chrome
-     */
+  get isInstalledLocally() {
+    return existsSync(this.path);
+  }
 
+  download() {
+    return request.get(
+      "https://clients2.google.com/service/update2/crx?response=redirect&os=win&arch=x86-64&os_arch=x86-64&nacl_arch=x86-64&prod=chromecrx&prodchannel=unknown&prodversion=77.0.3865.90&acceptformat=crx2&x=id=fmkadmapgofadopljbjfkapdkoienihi%26uc",
+      { encoding: null },
+      (err, res, body) => {
+        if (err) {
+          this.error(err);
+        }
+        if (!Buffer.isBuffer(body)) {
+          this.download();
+        }
+        const file = createWriteStream(this.path);
+        file.write(body, err => {
+          if (err) {
+            this.error(err);
+          }
+          file.close();
+        });
+        unzip(this.path, this.folderPath).then(() => {
+          this.listener();
+          this.log(
+            "If you are unable to find the tabs for React Developer Tools in the Developer Tools, Reload your client (ctrl + r)"
+          );
+        });
+      }
+    );
+  }
+
+  startPlugin() {
     this.listener = this.listener.bind(this);
-    remote.getCurrentWindow().webContents.on('devtools-opened', this.listener);
+    this.download();
```

Comment:
This will call download on every startup, and while it's a good solution for keeping it up to date i don't feel that should be left as-is. Maybe make it conditional if not installed or something?